addon/modifiers/basic-dropdown-trigger.ts

```diff
@@ -0,0 +1,138 @@
+import Modifier from 'ember-modifier';
+import { action } from '@ember/object';
+import hasMoved from '../utils/has-moved';
+import { Dropdown } from '../components/basic-dropdown';
+
+interface Args {
+  positional: []
+  named: {
+    dropdown: Dropdown
+    eventType?: 'click' | 'mousedown'
+    stopPropagation?: boolean
+  }
+}
+
+export default class DropdownTriggerModifier extends Modifier<Args> {
+  private toggleIsBeingHandledByTouchEvents: boolean = false
+  private touchMoveEvent?: TouchEvent
+
+  didInstall() {
+    if(!this.element.getAttribute('role')) this.element.setAttribute('role', 'button');
+
+    this.element.addEventListener('click', this.handleMouseEvent);
+    this.element.addEventListener('mousedown', this.handleMouseEvent);
+    this.element.addEventListener('keydown', this.handleKeyDown);
+    this.element.addEventListener('touchstart', this.handleTouchStart);
+    this.element.addEventListener('touchend', this.handleTouchEnd);
+  }
+
+  didReceiveArguments() {
+    const { dropdown } = this.args.named
+
+    this.element.setAttribute('data-ebd-id', `${dropdown?.uniqueId}-trigger`);
+    this.element.setAttribute('aria-owns', `ember-basic-dropdown-content-${dropdown?.uniqueId}`);
+    this.element.setAttribute('aria-expanded', dropdown?.isOpen ? 'true' : 'false');
+
+    if (dropdown?.disabled) {
+      // only remove if tabindex is not a custom value
+      if (this.element.getAttribute('tabindex') === '0') this.element.removeAttribute('tabindex');
+      this.element.setAttribute('aria-disabled', 'true');
+    } else {
+      // only set to 0 if tab index is not a custom value
+      if (this.element.getAttribute('tabindex') === null) this.element.setAttribute('tabindex', '0');
+      this.element.setAttribute('aria-disabled', 'false');
+    }
+  }
+
+  willRemove() {
+    if (typeof document !== 'undefined') document.removeEventListener('touchmove', this._touchMoveHandler);
```

Comment:
not sure why we have this `typeof document !== 'undefined'` check when optional chaining should suffice `document?.removeEventListener('touchmove', this._touchMoveHandler);`

I figured it might be a browser compatiblity thing?